Task 2/Task 2.1/Task 2.1.2/CustomPaintConsoleUI.cs

```diff
@@ -0,0 +1,225 @@
+                "\t3. Кольцо" + Environment.NewLine +
+                "\t4. Линия" + Environment.NewLine +
+                "\t5. Треугольник" + Environment.NewLine +
+                "\t6. Четырехугольник";
+        } 
+        public void StartMenu()
+        {
+            SwitchUserMenu();
+
+            string select = "";
+            while(select != "5")
+            {
+                //Console.Clear();
+                Console.WriteLine(_textStartMenu, _usersBD[currentUser].Name);
+
+                Console.Write("Ввод: ");
+                select = Console.ReadLine();
+
+                switch (select)
+                {
+                    case "1":
+                        AddFigureMenu();
+                        break;
+                    case "2":
+                        ShowFigures();
+                        break;
+                    case "3":
+                        ClearFigures();
+                        break;
+                    case "4":
+                        SwitchUserMenu();
+                        break;
+                }
+
+            }
+        }
+
+        private void SwitchUserMenu()
+        {
+            Console.Clear();
+
+            Console.WriteLine("Cписок пользователей:");
+            if (_usersBD.Count > 0)
+            {
+                foreach (var item in _usersBD)
+                {
+                    Console.WriteLine("{0, 5}{1, 10}", item.Key, item.Value.Name);
+                }
+            }
+            else
+            {
+                Console.WriteLine("\tТут пока что пусто.");
+            }
+
+            Console.WriteLine(_textSwitchUser);
+
+            string select = "";
+            do
+            {
+                Console.Write("Выберите действие: ");
+                select = Console.ReadLine();
+                switch (select)
+                {
+                    case "1":
+                        AddUser();
+                        break;
+                    case "2":
+                        if(_usersBD.Count > 0)
+                            SwitchUser();
+                        break;
+                    default:
+                        Console.WriteLine("Вы ввели что-то не то");
+                        break;
+                }
+            } while ((select != "1" && select != "2") || (select == "2" && _usersBD.Count == 0));
+
+            Console.Clear();
+        }
+
+        private void AddFigureMenu()
+        {
+            Console.WriteLine(_textAddFigureMenu);
+
+            int selectedFigure;
+            bool errorFlag = false;
+            do
+            {
+                if (errorFlag)
+                {
+                    Console.WriteLine("Такой фигуры нет");
+                }
+                errorFlag = true;
+
+                Console.Write("Выберите фигуру: ");
+
+            } while (!(int.TryParse(Console.ReadLine(), out selectedFigure))||
+                     (selectedFigure < 1 || selectedFigure > 6));
```

Comment:
to make it less sophisticated
```suggestion
            } while (!(int.TryParse(Console.ReadLine(), out selectedFigure)
                     && selectedFigure > 1 && selectedFigure < 6));
```
Better would be to read it inside loop block and make a bool variable `retry` - this way you can even get rid of `errorFlag`